Source/net_defs.h

```diff
@@ -0,0 +1,276 @@
+// compatibility with Chocolate Doom servers. Higher-numbered enum values
+// will be preferred when negotiating a protocol for the client and server
+// to use, so the order matters.
+// NOTE: The values in this enum do not have any special value outside of
+// the program they're compiled in. What matters is the string representation.
+typedef enum
+{
+    // Protocol introduced with Chocolate Doom v3.0. Each compatibility-
+    // breaking change to the network protocol will produce a new protocol
+    // number in this enum.
+    NET_PROTOCOL_CHOCOLATE_DOOM_0,
+
+    // Add your own protocol here; be sure to add a name for it to the list
+    // in net_common.c too.
+
+    NET_NUM_PROTOCOLS,
+    NET_PROTOCOL_UNKNOWN,
+} net_protocol_t;
+
+// packet types
+
+typedef enum
+{
+    NET_PACKET_TYPE_SYN,
+    NET_PACKET_TYPE_ACK, // deprecated
+    NET_PACKET_TYPE_REJECTED,
+    NET_PACKET_TYPE_KEEPALIVE,
+    NET_PACKET_TYPE_WAITING_DATA,
+    NET_PACKET_TYPE_GAMESTART,
+    NET_PACKET_TYPE_GAMEDATA,
+    NET_PACKET_TYPE_GAMEDATA_ACK,
+    NET_PACKET_TYPE_DISCONNECT,
+    NET_PACKET_TYPE_DISCONNECT_ACK,
+    NET_PACKET_TYPE_RELIABLE_ACK,
+    NET_PACKET_TYPE_GAMEDATA_RESEND,
+    NET_PACKET_TYPE_CONSOLE_MESSAGE,
+    NET_PACKET_TYPE_QUERY,
+    NET_PACKET_TYPE_QUERY_RESPONSE,
+    NET_PACKET_TYPE_LAUNCH,
+    NET_PACKET_TYPE_NAT_HOLE_PUNCH,
+} net_packet_type_t;
+
+typedef enum
+{
+    NET_MASTER_PACKET_TYPE_ADD,
+    NET_MASTER_PACKET_TYPE_ADD_RESPONSE,
+    NET_MASTER_PACKET_TYPE_QUERY,
+    NET_MASTER_PACKET_TYPE_QUERY_RESPONSE,
+    NET_MASTER_PACKET_TYPE_GET_METADATA,
+    NET_MASTER_PACKET_TYPE_GET_METADATA_RESPONSE,
+    NET_MASTER_PACKET_TYPE_SIGN_START,
+    NET_MASTER_PACKET_TYPE_SIGN_START_RESPONSE,
+    NET_MASTER_PACKET_TYPE_SIGN_END,
+    NET_MASTER_PACKET_TYPE_SIGN_END_RESPONSE,
+    NET_MASTER_PACKET_TYPE_NAT_HOLE_PUNCH,
+    NET_MASTER_PACKET_TYPE_NAT_HOLE_PUNCH_ALL,
+} net_master_packet_type_t;
+
+// Settings specified when the client connects to the server.
+
+// typedef struct
+// {
+//     int gamemode;
+//     int gamemission;
+//     int lowres_turn;
+//     int drone;
+//     int max_players;
+//     int is_freedoom;
+//     sha1_digest_t wad_sha1sum;
+//     sha1_digest_t deh_sha1sum;
+//     int player_class;
+// } net_connect_data_t;
+
+// Game settings sent by client to server when initiating game start,
+// and received from the server by clients when the game starts.
+
+typedef struct
+{
+    int ticdup;
+    int extratics;
+    int deathmatch;
+    int episode;
+    int nomonsters;
+    int fast_monsters;
+    int respawn_monsters;
+    int map;
+    int skill;
+    int gameversion;
+    int lowres_turn;
+    int new_sync;
+    int timelimit;
+    int loadgame;
+    int random;  // [Strife only]
+
+    // These fields are only used by the server when sending a game
+    // start message:
+
+    int num_players;
+    int consoleplayer;
+
+    // Hexen player classes:
```

Comment:
Please clean these files off of everything that is not needed to get midiproc running.